hazelcast/src/main/java/com/hazelcast/internal/memory/impl/DirectMemoryBits.java

```diff
@@ -37,7 +41,7 @@ public static char readChar(long address, boolean bigEndian) {
     }
 
     public static char readChar(byte[] buffer, int pos, boolean bigEndian) {
-        return readChar(buffer, (long) (BYTE_ARRAY_BASE_OFFSET + pos), bigEndian);
+        return readChar(BYTE_BUFFER_MEMORY_ACCESSOR, buffer, pos, bigEndian);
```

Comment:
I think, `Bits` itself should pass `BYTE_BUFFER_MEMORY_ACCESSOR` to `DirectMemoryBits` by calling `char readChar(MemoryAccessor memoryAccessor, Object base, long offset, boolean bigEndian)` and by default `DirectMemoryBits` should use `UNSAFE_MEMORY_ACCESSOR` even though passed object is `byte[]`